cEP-0022.md

```diff
@@ -0,0 +1,461 @@
+  the
+  `.quickfile`, weights of the other values to bear settings will be appended.
+- Successive runs of **Brute Force** will assume that it is being run
+  on the results after a few `--lite-mode` runs and will only run
+  for the settings
+  provided in the `.quickfile` appending to the weights.
+
+### **The `--smart-mode`**
+
+It is clear that successive runs of the `--lite-mode` will overestimate
+the weights in the `.quickfile`, so it seems advantageous to run `--lite-mode`
+several times and then run the **Brute Force** for the settings dropped by
+`--lite-mode` which are now in `.quickfile` to generate correct value of the
+weight. This combination of `--lite-mode` and **Brute Force** is done
+collectively by `--smart-mode`.
+
+Every run will have the annoying question in the end whether the user wants
+to answer
+a few questions about the project to get better results unless changed upon
+by the provided tags. *(For more details check out the tags section)*
+
+# Learning from real world projects
+
+## Need to perform this task
+
+There can be endless number of possibilities and assumptions:
+
+- Files having similar kind of names can be having different codestyles.
+- Files in a given directory can be following different codestyles.
+- Different parts of files maybe even following a different codestyle depending
+  on function name which can also be handled by placing `ignore`s at
+  appropriate places in the file dict.
+
+  Such kind of functionality can only be
+  added to the bears themselves, but instead we stop our train of thought over
+  here.
+
+## Making Quickstart recognize common patterns
+
+We save ourselves the additional work for building some feature that may
+hardly ever be used by any org, instead we stop assumming and start
+implementing and learning from our available resources.
+
+We now choose a list of orgs for which we try to test our `green-mode`
+ourselves. We check whether brute force is taking too much time or whether
+the `--lite-mode` or the `--smart-mode` is producing the correct results
+(i.e. *green* results). We
+choose these orgs in such a way that they are very famous or very well known or
+act as upstream repositories for a huge amount of orgs. This way a very large
+number of orgs may be mimicking the code style of these orgs itself and if
+our further optimizations can fix the problem of generating *green* config
+files for these organizations, we are in turn solving the problem for these
+other orgs/communities at the same time.
+
+We learn manually from these orgs, what combinations of these bear-setting
+values are they using, whether they are using it all over their project or only
+in certain scenarious, whether they have different codestyles in different
+directories or different nomenclature of file patterns require a different
+set of settings in their project. We give `coala-quickstart`, the ability
+to recognizing these scenarios for further
```

Comment:
Don't we need more detailed data to deduce such patterns, or is it just a manual process by observation?

Reply:
manual currently

Reply:
>We learn manually from these orgs, what combinations of these bear-setting